File: 1.1/hierarchical_model_1-1.py
import sys, math
import numpy as np
import pystan
import matplotlib
matplotlib.use('Agg')
import matplotlib.pyplot as plt

def simulate():
  print '\tGenerating simulated data...'

  ###################################################
  ## Your code here


  ## Your code here
  ###################################################

  # Construct dictionary to pass to Stan
  dataset = {
    'N': N,
    'D': D,
    'observations': observations,
  }
  return dataset


def inference(dataset):
  print '\tPerforming inference...'
  NUM_ITER = 2000
  WARMUP = 200
  NUM_CHAINS = 4
  NUM_CORES = 4
  STAN_FN = 'hierarchical_model_1-1.stan'

  # import pdb; pdb.set_trace()
  fit = pystan.stan(file = STAN_FN, 
                    data = dataset, 
                    iter = NUM_ITER, 
                    warmup = WARMUP, 
                    chains = NUM_CHAINS, 
                    n_jobs = NUM_CORES)
  print(fit)

  fit.plot()
  plt.tight_layout()
  plt.savefig('fit.png')
  return


def main():
  dataset = simulate()
  inference(dataset)
  return


if __name__ == '__main__':
  main()

File: 1.1/hierarchical_model_1-1.R
library('rstan');

#########################################################
# Simulate data
#########################################################
## Your code here




## Your code here

#########################################################
# Run inference
#########################################################

# Construct dataset to pass to Stan  
dataset <- list(N = N, 
                D = D,
                observations = obs);

fit <- stan(file = 'hierarchical_model_1-1.stan', 
            data = dataset, 
            iter = 1000, 
            chains = 4);
print(fit);

png(filename = 'fit_rstan.png')
plot(fit);
dev.off();

fe <- extract(fit);
print( tail(fe$normal_mus, n = 1) );